src/proto/cloudevents/protocloudevent.rs

```diff
@@ -23,6 +23,58 @@ use crate::cloudevents::cloud_event::CloudEventAttributeValue;
 use crate::cloudevents::cloud_event::Data as CloudEventData;
 use crate::cloudevents::CloudEvent as CloudEventProto;
 
+use crate::uprotocol::umessage::UMessage;
+use crate::uri::serializer::{LongUriSerializer, UriSerializer};
```

Comment:
IMHO it'd be nice if we're only using `LongUriSerializer` to only have to use that one, i.e. only have to:
```rust
use crate::uri::serializer::{LongUriSerializer};
```

I think we can achieve that by [modifying this line](https://github.com/eclipse-uprotocol/up-rust/blob/ae311dd9d170d7030344fb020a9ed83342f6b0d6/src/uri/serializer/longuriserializer.rs#L19):
```rust
use crate::uri::serializer::{SerializationError, UriSerializer};
```
to
```rust
pub use crate::uri::serializer::{SerializationError, UriSerializer};
```

Reply:
Good catch!

Reply:
Addendum - this doesn't really work for some reason, dependent code still won't compile without explicitly use-ing the Trait implementation. I guess I'll revert this change for now.

Reply:
My bad! Looks like this is not possible in Rust. We have to bring the trait into scope explicitly with a `use`. Seems this is a conscious design decision to help when multiple traits define things with the same names and helps readability to understand where a function could be coming from.

Reply:
heh - forget all that, found the problem. We will have a try_from.